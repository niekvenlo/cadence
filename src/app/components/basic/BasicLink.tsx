"use client";

import Link from "next/link";
import { usePathname } from "next/navigation";

type Props = {
  children: string;
  href: string;
};

export default function BasicLink({ href, children }: Props) {
  const isActive = usePathname() === href;
  return (
    <Link data-active={isActive || null} href={href}>
      {children}
    </Link>
  );
}
