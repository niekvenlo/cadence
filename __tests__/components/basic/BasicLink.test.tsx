import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import BasicLink from "../../../src/app/components/basic/BasicLink";

jest.mock("next/navigation", () => ({
  usePathname: () => "/active",
}));

describe("BasicLink", () => {
  it("renders a basic link", () => {
    render(<BasicLink href="/not-active">label</BasicLink>);
    const link = screen.getByText("label");

    expect(link).toBeInTheDocument();
    expect(link).not.toHaveAttribute("data-active");
    expect(link).toMatchSnapshot();
  });
  it("renders a basic active link", () => {
    render(<BasicLink href="/active">label</BasicLink>);
    const link = screen.getByText("label");

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("data-active");
    expect(link).toMatchSnapshot();
  });
});
