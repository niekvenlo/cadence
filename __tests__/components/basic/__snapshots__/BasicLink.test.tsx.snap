// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLink renders a basic active link 1`] = `
<a
  data-active="true"
  href="/active"
>
  label
</a>
`;

exports[`BasicLink renders a basic link 1`] = `
<a
  href="/not-active"
>
  label
</a>
`;
